controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -112,7 +112,11 @@ func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (controller.O
 				a.logger.Error(err, "Failed to create pipelineRun for snapshot, environment and scenario")
 				return controller.RequeueWithError(err)
 			}
-			metrics.RegisterPipelineRunWithEphemeralEnvStarted(a.snapshot.CreationTimestamp, metav1.Now())
+
+			// measure only non-reruns
+			if _, ok := gitops.GetIntegrationTestRunLabelValue(a.snapshot); !ok {
```

Comment:
```suggestion
			if _, ok := gitops.GetIntegrationTestRunLabelValue(a.snapshotEnvironmentBinding); !ok {
```
If I'm reading this correctly, we should check the labels on the SnapshotEnvironmentBinding instead of Snapshot, right?

Reply:
yes :(, fixed, I should really try to write a unittest for it.